### PR TITLE
WebGL Web3Auth Login Nikola's fix

### LINKED ISSUE #553: WebGL build of the Web3Auth is not working
**Describe the bug**
WebGL build of the SDK is not working.

**To Reproduce**
1. Create WebGL build of a SampleScene
2. Login with Web3Auth
3. The redirection is happening on the same page so on redirecting back the game loosing state and reloads.

**Expected behavior**
Game loads the next scene after Web3Login.

### PR DESCRIPTION
**Pull request by kalambet**
Closes #553

### CODE REVIEW

**Comment by kantagara** [comment]
@kalambet 
image
We need these three as well.

**Comment by kantagara** [comment]
@kalambet they're in.

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 1]
```diff
@@ -313,6 +346,8 @@ public void logout(Uri redirectUrl = null, string appState = null)
     private void authorizeSession()
     {
         string sessionId = KeyStoreManagerUtils.getPreferencesData(KeyStoreManagerUtils.SESSION_ID);
+        Debug.Log($"in auth check {sessionId}");
+        Debug.Log($"Session check is: {!string.IsNullOrEmpty(sessionId)}");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:350]: We can likely remove these
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 2]
```diff
@@ -313,6 +346,8 @@ public void logout(Uri redirectUrl = null, string appState = null)
     private void authorizeSession()
     {
         string sessionId = KeyStoreManagerUtils.getPreferencesData(KeyStoreManagerUtils.SESSION_ID);
+        Debug.Log($"in auth check {sessionId}");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:349]: We can likely remove these
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 3]
```diff
@@ -396,15 +431,18 @@ private void sessionTimeOutAPI()
                         {
                             if (result != null)
                             {
+                                Debug.Log("Coroutine logout result not null");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:434]: We can likely remove these
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 4]
```diff
@@ -396,15 +431,18 @@ private void sessionTimeOutAPI()
...
                                 try
                                 {
                                     KeyStoreManagerUtils.deletePreferencesData(KeyStoreManagerUtils.SESSION_ID);
                                     KeyStoreManagerUtils.deletePreferencesData(web3AuthOptions.loginConfig?.Values.First()?.verifier);
+                                    Debug.Log("DEleted things probably");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:439]: We can likely remove these
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 5]
```diff
@@ -396,15 +431,18 @@ private void sessionTimeOutAPI()
...
 
                                     this.Enqueue(() => this.onLogout?.Invoke());
                                 }
                                 catch (Exception ex)
                                 {
+                                    Debug.Log("Errored deleting");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:445]: We can likely remove these
```suggestion
```

File: src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Prefabs/Logout.cs [hunk 6]
```diff
@@ -6,6 +6,7 @@ public class Logout : MonoBehaviour
 {
     public async void OnLogout()
     {
+        Debug.Log("Trying the other login");
```
Comment (RyRy79261) [inline src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Prefabs/Logout.cs:9]: We can likely remove these
```suggestion
```

**Review by RyRy79261 — CHANGES_REQUESTED** [review]

**Comment by kalambet** [comment]
@RyRy79261 I would rather not touch stuff in `[Web3AuthSDK]` folder more than necessary. It is a pure import from Web3Auth repo and the less changes we have the easier it would be to merge with the upstream.

I would rather go with something like:
```csharp
 #if UNITY_EDITOR
 Debug.logger.logEnabled = true;
 #else
 Debug.logger.logEnabled = false;
 #endif
 ```

**Comment by RyRy79261** [comment]
@kalambet So those lines I mentioned are actually my debug lines I intended to remove on my branch before they were cherry picked to Nikola's fix 😅

**Comment by kalambet** [comment]
Lol, I see. That make sense 🤣
Will update PR shortly!

**Comment by sneakzttv** [comment]
awaiting update :D

**Review by kantagara — APPROVED** [review]

**Comment by kalambet** [comment]
Just in case. I left Gelato 1Balance API key in the sources intentional so you can test all the buttons in a Demo Scene.

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs [hunk 7]
```diff
@@ -300,6 +354,7 @@ public void logout(Dictionary<string, object> extraParams)
 
     public void logout(Uri redirectUrl = null, string appState = null)
     {
+        Debug.Log($"redirectUrl: {redirectUrl} appState:{appState}");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Plugins/Web3AuthSDK/Web3Auth.cs:357]: Could remove this, one of my debug lines
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs [hunk 8]
```diff
@@ -35,12 +38,16 @@ public Web3AuthWallet(Web3AuthWalletConfig config, IChainConfig chainConfig, IRp
         public async ValueTask WillStartAsync()
         {
             coreInstance = CreateCoreInstance();
-
+            string privateKeyString;
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs:41]: Could declare this on line 50, left over from a different logic
```suggestion
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs [hunk 9]
```diff
@@ -35,12 +38,16 @@ public Web3AuthWallet(Web3AuthWalletConfig config, IChainConfig chainConfig, IRp
...
-            coreInstance.login(config.LoginParams);
-            var privateKeyString = await loginTcs.Task;
-            coreInstance.onLogin -= Web3Auth_OnLogin;
+
+            if (config.LoginParams != null)
+            {
+                coreInstance.login(config.LoginParams);
+            }
+
+            privateKeyString = await loginTcs.Task;
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs:50]:
```suggestion
            var privateKeyString = await loginTcs.Task;
```

File: src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs [hunk 10]
```diff
@@ -54,6 +61,7 @@ public async ValueTask WillStartAsync()
         public async ValueTask WillStopAsync()
         {
             TaskCompletionSource<object> logoutTcs = new();
+            Debug.Log("Logging out via WillStop");
```
Comment (RyRy79261) [inline src/UnityPackages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWallet.cs:64]: Could remove 
```suggestion
```

**Review by RyRy79261 — COMMENTED** [review]
Minor changes, left overs from my inprogress commits

**Review by RyRy79261 — APPROVED** [review]
Tested in Webgl, works perfectly

**Review by sneakzttv — APPROVED** [review]

**Review by juans-chainsafe — APPROVED** [review]
Awesome! working on WebGL in both templates as well.

**Pull request merged**